exodus_lambda/functions/origin_request.py

```diff
@@ -1,27 +1,52 @@
 import json
+import time
 import urllib
-from datetime import datetime, timezone
+from datetime import datetime, timedelta, timezone
 
 import boto3
-from cdn_definitions import load_data
+import cachetools
 
 from .base import LambdaBase
 
 
 class OriginRequest(LambdaBase):
-    def __init__(
-        self, conf_file="lambda_config.json", definitions_source=None
-    ):
+    def __init__(self, conf_file="lambda_config.json"):
         super().__init__("origin-request", conf_file)
         self._db_client = None
-        self._definitions_source = definitions_source
         self._definitions = None
+        self._cache = cachetools.TTLCache(
+            maxsize=1,
+            ttl=timedelta(
+                minutes=self.conf.get("config_cache_ttl", 2)
+            ).total_seconds(),
+            timer=time.monotonic,
+        )
 
     @property
     def definitions(self):
-        if self._definitions is None:
-            self._definitions = load_data(source=self._definitions_source)
-        return self._definitions
+        if self._cache.get("exodus-config") is None:
+            table = self.conf["config_table"]["name"]
+
+            query_result = self.db_client.query(
+                TableName=table,
+                Limit=1,
+                ScanIndexForward=False,
+                KeyConditionExpression="config_id = :id and from_date <= :d",
+                ExpressionAttributeValues={
+                    ":id": {"S": "exodus-config"},
+                    ":d": {
+                        "S": str(
+                            datetime.now(timezone.utc).isoformat(
+                                timespec="milliseconds"
+                            )
+                        )
+                    },
+                },
+            )
+            if query_result["Items"]:
+                item = query_result["Items"][0]
+                self._cache["exodus-config"] = json.loads(item["config"]["S"])
+        return self._cache.get("exodus-config")
```

Comment:
Minor: it would generally be a bad practice to write code like this when using a cache from cachetools:

```
cache["some-key"] = ...
return cache.get("some-key")
```

Problem: you don't know if the cache has become invalidated again between line 1 and 2.  The cache invalidates itself without warning.  So, for example, for a time-based cache it can expire between 1 and 2 due to the passage of time; for a cache with a maxsize it can expire between 1 and 2 because another thread ran meanwhile and inserted something else into the cache bumping it past the maxsize and causing evictions.  Just because you put something in the cache at one line of code doesn't guarantee that it's still present when the next line of code executes.

The safe way to write the code would be more like:

```
out = ...
cache["some-key"] = out
return out
```

So you write to the cache, but then you don't assume that the value you just wrote can be pulled back out.

I'm calling it minor since the specific choice of cache, and absence of threads in this code should make it very unlikely for any genuine issue here.